chain/bitcoind_client.go

```diff
@@ -204,36 +222,27 @@ func (c *BitcoindClient) NotifySpent(outPoints []*wire.OutPoint) error {
 	return nil
 }
 
-// NotifyTxIDs updates the watch list with the passed TxIDs.
-func (c *BitcoindClient) NotifyTxIDs(txids []chainhash.Hash) error {
-	c.NotifyBlocks()
-	select {
-	case c.rescanUpdate <- txids:
-	case <-c.quit:
-		return ErrBitcoindClientShuttingDown
-	}
-
-	return nil
-}
-
-// NotifyBlocks enables notifications.
+// NotifyBlocks allows the chain backend to notify the caller whenever a block
+// is connected or disconnected.
+//
+// NOTE: This is part of the chain.Interface interface.
 func (c *BitcoindClient) NotifyBlocks() error {
 	atomic.StoreUint32(&c.notifyBlocks, 1)
 	return nil
 }
 
-// notifying returns true if notifications have been turned on; false otherwise.
-func (c *BitcoindClient) notifying() bool {
+// shouldNotifyBlocks determines whether the client should send block
+// notifications to the caller.
+func (c *BitcoindClient) shouldNotifyBlocks() bool {
 	return (atomic.LoadUint32(&c.notifyBlocks) == 1)
 }
 
-// LoadTxFilter updates the transaction watchlists for the client. Acceptable
-// arguments after `reset` are any combination of []btcutil.Address,
-// []wire.OutPoint, []*wire.OutPoint, []chainhash.Hash, and []*chainhash.Hash.
-func (c *BitcoindClient) LoadTxFilter(reset bool,
-	watchLists ...interface{}) error {
+// LoadTxFilter uses the given outpoints and addresses as filters to what we
+// should match transactions against to determine if they are relevant to the
+// client. The reset argument is used to reset the current filters.
+func (c *BitcoindClient) LoadTxFilter(reset bool, outpoints []wire.OutPoint,
```

Comment:
This is too restrictive. As is, things beyond outpoint and addresses can be sent.

Reply:
Fixed.